### PR TITLE
fix(completion): remove bound insert of type in trait

### PR DESCRIPTION
**Pull request by bvanjoi**
Fixed https://github.com/rust-lang/rust-analyzer/issues/13819

### CODE REVIEW

File: crates/ide-completion/src/tests/expression.rs [hunk 1]
```diff
@@ -652,6 +652,26 @@ fn main() {
...
+    $0
+}
+",
+        expect![[r"
+    ta type O =
+    kw crate::
+    kw self::
+    "]],
+    )
+}
```
Comment (Veykril) [inline crates/ide-completion/src/tests/expression.rs:673]: This isn't an expression test, this test should go into `item_list.rs`

File: crates/ide-completion/src/completions/item_list/trait_impl.rs [hunk 2]
```diff
@@ -265,10 +265,24 @@ fn add_type_alias_impl(
...
 
             let start = transformed_ty.syntax().text_range().start();
-            let Some(end) = transformed_ty
-                .eq_token()
+
+            let end = if let Some(end) = transformed_ty
+                .syntax()
+                .children_with_tokens()
+                .find(|it| it.kind() == T![:])
                 .map(|tok| tok.text_range().start())
```
Comment (Veykril) [inline crates/ide-completion/src/completions/item_list/trait_impl.rs:273]:
```suggestion
            let end = if let Some(end) = transformed_ty.colon_token()
                .map(|tok| tok.text_range().start())
```

Comment (Veykril) [reply crates/ide-completion/src/completions/item_list/trait_impl.rs:273]: This change needs a `HasTypeBounds` trait import

File: crates/ide-completion/src/tests/expression.rs (hunk 1 above)
Comment (bvanjoi) [reply crates/ide-completion/src/tests/expression.rs:673]: I would remove this test case as it seems a bit redundant

File: crates/ide-completion/src/completions/item_list/trait_impl.rs (hunk 2 above)
Comment (bvanjoi) [reply crates/ide-completion/src/completions/item_list/trait_impl.rs:273]: thanks for your tips

File: crates/ide-completion/src/render.rs [hunk 3]
```diff
@@ -594,6 +594,165 @@ fn main() { Foo::Fo$0 }
         );
     }
 
+    #[test]
+    fn type_in_impl_trait() {
```
[the remark this answers is not in the document]
Comment (Veykril) [reply crates/ide-completion/src/render.rs:598]: These tests should still go into `crates/ide-completion/src/tests/item_list.rs`

[the remark this answers is not in the document]
Comment (bvanjoi) [reply crates/ide-completion/src/render.rs]: done

**Comment by Veykril** [comment]
Thanks!
@bors r+

**Pull request merged**